From: Rob Evers <revers@redhat.com>
Date: Tue, 16 Feb 2010 19:37:54 -0500
Subject: lpfc: Fix a merge issue
Message-id: <20100216193715.8604.58002.sendpatchset@localhost.localdomain>
Patchwork-id: 23293
O-Subject: [RHEL5.5 3/7] lpfc: Fix a merge issue
Bugzilla: 564506
RH-Acked-by: Mike Christie <mchristi@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

Fix a merge issue

https://bugzilla.redhat.com/show_bug.cgi?id=564506
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_scsi.c b/drivers/scsi/lpfc/lpfc_scsi.c
index f190216..6c54c28 100644
--- a/drivers/scsi/lpfc/lpfc_scsi.c
+++ b/drivers/scsi/lpfc/lpfc_scsi.c
@@ -1140,13 +1140,12 @@ lpfc_scsi_prep_dma_buf_s4(struct lpfc_hba *phba, struct lpfc_scsi_buf *lpfc_cmd)
 		sgl->word2 = cpu_to_le32(sgl->word2);
 		sgl += 1;
 
-		bf_set(lpfc_sli4_sge_len, sgl, scsi_cmnd->request_bufflen);
 		sgl->addr_lo = cpu_to_le32(putPaddrLow(physaddr));
 		sgl->addr_hi = cpu_to_le32(putPaddrHigh(physaddr));
 		bf_set(lpfc_sli4_sge_last, sgl, 1);
 		bf_set(lpfc_sli4_sge_offset, sgl, dma_offset);
 		sgl->word2 = cpu_to_le32(sgl->word2);
-		sgl->word3 = cpu_to_le32(sgl->word3);
+		sgl->sge_len = cpu_to_le32(scsi_cmnd->request_bufflen);
 
 		lpfc_cmd->nonsg_phys = physaddr;
 	} else {
